sklearn/utils/estimator_checks.py

```diff
@@ -1190,6 +1190,30 @@ def check_estimators_pickle(name, estimator_orig):
         unpickled_result = getattr(unpickled_estimator, method)(X)
         assert_allclose_dense_sparse(result[method], unpickled_result)
 
+    # check if input with nans pickles properly
+    if name in ALLOW_NAN:
+        # set random 10 elements to be corrupted
+        X.ravel()[np.random.choice(X.size, 10, replace=False)] = np.nan
+        estimator.fit(X, y)
+
+        nan_result = dict()
+        for method in check_methods:
+            if hasattr(estimator, method):
+                nan_result[method] = getattr(estimator, method)(X)
+
+        # pickle and unpickle!
+        pickled_estimator = pickle.dumps(estimator)
+        if estimator.__module__.startswith('sklearn.'):
+            assert_true(b"version" in pickled_estimator)
+        unpickled_estimator = pickle.loads(pickled_estimator)
+
+        for method in nan_result:
+            unpickled_nan_result = getattr(unpickled_estimator, method)(X)
+            if name == 'ChainedImputer':
+                assert_allclose_dense_sparse(nan_result[method], unpickled_nan_result, rtol=1e-5, atol=0.1)
```

Comment:
Shouldn't this pas without the increased tolerance if this aims to fix https://github.com/scikit-learn/scikit-learn/issues/11462 ? or am I missing something?